### PR TITLE
Add Browserify task and ES6 transpire option

### PR DESCRIPTION
**Pull request by chuckcarpenter**
https://jira.nationalgeographic.com/browse/SYTO-80

This pull request adds the Browserify task to Grunt and an option for ES6 transform using [6to5](https://6to5.org). This will allow us to choose to write ES6 and have it work in the browser now, while being able to remove these steps when ES6 support is released across all supported browsers. 

To test:
If you don't have the generator, you'll want to close the repo then:
`npm install && npm link`
This will setup the generator and create a symlink as if it was globally installed. 
Then, create a test directory, CD into it and run `yo ngs` and follow the prompts. There are not any JS files that get loaded in to test the task.

### CODE REVIEW

File: app/templates/_grunt/browserify.coffee [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+module.exports =
+    dist:
+        files:
+            'dist/app.js': ['js/**/*.{jsx,js}']
+
+        options:
+            browserifyOptions:
+                debug: true, # sourcemaps
+                extensions: ['.jsx', '.js'] # consider jsx files as modules
+            transform: <% if (ES6 == true) { %>[to5ify]<% } %>
```
Comment (asabaylus) [inline app/templates/_grunt/browserify.coffee]: I think there's a typo... I noticed the the 6to5 docs also have the same typo.
to5ify ==> 6to5ify

Comment (chuckcarpenter) [inline app/templates/_grunt/browserify.coffee]: Not if you set the var as that. `<% if (ES6 == true) { %>to5ify = require '6to5ify'<% } %>`

Comment (asabaylus) [inline app/templates/_grunt/browserify.coffee]: doh! :+1:

**Comment by asabaylus** [comment]
I think this looks good, ship it! :+1: :+1:

**Pull request merged**